src/components/Firebase/firebase.js

```diff
@@ -2,129 +2,260 @@ import firebase from 'firebase';
-    var questionNum = 1;
-    quiz.forEach(QASet => {
-      quizRef.child(questionNum).set(QASet);
-      questionNum += 1;
-    });
+  createQuiz = (name, cb) => {
+    // create an empty round array
+    // 11 questions in case of bonus
+    const round = [];
+    for (let i = 0; i < 11; i++) {
+      round.push({
+        q: '',
+        a: '',
+        img: '',
+      });
+    }
 
-    callback();
+    // create a new quiz with a random id
+    this.db.collection('quizzes').add({
+      name,
+      date: this.getCurrentFormattedDate(),
+      round1: round,
+      round2: round,
+      round3: round,
+    })
+      .then((ref) => {
+        cb(ref.id);
+      });
   };
 
   /**
-   * @param {number} teamNumber
-   * @param {array} answers
+   * Delete a quiz by its id
+   * @param {String} quizId
    * @param {function} callback
    */
-  saveAnswersForTeam = (teamNumber, answers, callback) => {
-    let teamRef = firebase.database().ref()
-      .child(this.getCurrentFormattedDate())
-      .child('teams')
-      .child(teamNumber);
-
-    answers.forEach((answer, index) => {
-      teamRef.child(index).set(answer);
-    });
+  deleteQuiz = (id, cb) => {
+    this.db.collection(`quizzes`).doc(id).delete().then(() => cb());
+  };
 
-    callback();
+
+  /**
+   * save a whole quiz
+   * it's essential that the quiz is the whole quiz object
+   * @param {String} quizId
+   * @param {Object} quiz
+   * @param {function} callback
+   */
+  saveQuiz = (id, quiz, cb) => {
+    const docRef = this.db.collection('quizzes').doc(id);
+    docRef.set(quiz).then(() => cb());
   };
 
   /**
-   * @param {Number} teamNumber
+   * returns all quizzes in the database
    * @param {function} callback
    */
-  getAnswersForTeam = (teamNumber, callback) => {
-    let teamRef = firebase.database().ref()
-      .child(this.getCurrentFormattedDate())
-      .child('teams')
-      .child(teamNumber);
-
-    teamRef.once('value')
-      .then(snapshot => {
-        if (!snapshot.exists()) {
-          return;
+  getQuizzes = (cb) => {
+    const quizzes = [];
+    this.db.collection('quizzes').get()
+      .then((snap) => {
+        snap.forEach((doc) => {
+          quizzes.push({
+            id: doc.id,
+            data: doc.data(),
+          });
+        });
+        cb(quizzes);
+      })
+      .catch((err) => {
+        console.error('Error getting documents', err);
+      });
+  };
+
+  /**
+   * Get a quiz by its ID
+   * @param {String} quizID
+   * @param {function} callback
+   */
+  getQuiz = (id, cb) => {
+    this.db.collection('quizzes').doc(id).get()
+      .then((doc) => {
+        // if no doc, return no success so we can redirect
+        // else return the document data
+        if (!doc) {
+          cb({ success: false });
+        } else {
+          cb({ success: true, data: doc.data() });
         }
+      })
+      .catch((err) => {
+        console.error('Error getting document', err);
+        cb({ success: false });
```

Comment:
Client can't distinguish whether the `getQuiz` failed due to the document being empty or due to an actual server-side error, which could be something like loss of internet connection, which is actually unrelated to whether or not the document exists. I think the better practice would be to return no data if the document doesn't exist and have the client check that case.